v2/pkg/genruntime/base_types.go

```diff
@@ -84,21 +84,40 @@ type ARMOwnedMetaObject interface {
 // of empty string will result in the removal of that annotation.
 func AddAnnotation(obj MetaObject, k string, v string) {
 	annotations := obj.GetAnnotations()
-	if annotations == nil {
-		annotations = map[string]string{}
+	annotations = AddToMap(annotations, k, v)
+	obj.SetAnnotations(annotations)
+}
+
+// RemoveAnnotation removes the specified annotation from the object
+func RemoveAnnotation(obj MetaObject, k string) {
+	AddAnnotation(obj, k, "")
+}
+
+// AddLabel adds the specified label to the object.
+// Empty string labels are not allowed. Attempting to add a label with a value
+// of empty string will result in the removal of that label.
+func AddLabel(obj MetaObject, k string, v string) {
+	labels := obj.GetLabels()
+	labels = AddToMap(labels, k, v)
+	obj.SetLabels(labels)
+}
+
+func AddToMap(labels map[string]string, k string, v string) map[string]string {
```

Comment:
minor: Consider not exporting this, and changing the parameter name from `labels` or `annotations` to something generic like `m`?